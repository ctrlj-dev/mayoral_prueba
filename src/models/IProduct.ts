export default interface IProduct {
    id: number,
    name: string,
    price: number,
    discount?: number,
    colors?: string[]
    featuredImg: string
}