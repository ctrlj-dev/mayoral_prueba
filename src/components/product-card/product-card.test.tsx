import ProductCardComponent from './product-card.component';
import { render } from '@testing-library/react';
import IProduct from 'models/IProduct';
import React from 'react';

describe('ProductCardComponent specs', () => {
    test('It should display all the properties need by the component', () => {
        const product: IProduct = {
            "id": 0,
            "name": "Polo manga corta estampada",
            "price": 18.99,
            "discount": 20,
            "colors": [
                "'rojo'",
                "'verde'",
                "'az√∫l'"
            ],
            "featuredImg": "https://json.ctrlj.es/mayoral/img/polo-azul-marino.jpg"
        };

        //Arrange
        const featuredImg = 'https://json.ctrlj.es/mayoral/img/polo-azul-marino.jpg';
        //Act
        const { getByText } = render(
            <ProductCardComponent
                id={product.id}
                name={product.name}
                discount={product.discount}
                price={product.price}
                featuredImg={product.featuredImg}
                colors={product.colors}
            />)

        const elementName = getByText('Polo manga corta estampada');
        expect(elementName).not.toBeNull();
        expect(elementName.tagName).toEqual('H2')

        //Assert
    })
}) 