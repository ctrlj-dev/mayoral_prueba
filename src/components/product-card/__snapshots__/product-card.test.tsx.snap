// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard specs It should display all the properties need by the component 1`] = `
<DocumentFragment>
  <div
    class="productCard undefined"
  >
    <div
      class="productCardContainer"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
        />
        <img
          alt="Polo manga corta estampada"
          data-nimg="responsive"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fjson.ctrlj.es%2Fmayoral%2Fimg%2Fpolo-azul-marino.jpg&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fjson.ctrlj.es%2Fmayoral%2Fimg%2Fpolo-azul-marino.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fjson.ctrlj.es%2Fmayoral%2Fimg%2Fpolo-azul-marino.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fjson.ctrlj.es%2Fmayoral%2Fimg%2Fpolo-azul-marino.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fjson.ctrlj.es%2Fmayoral%2Fimg%2Fpolo-azul-marino.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fjson.ctrlj.es%2Fmayoral%2Fimg%2Fpolo-azul-marino.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fjson.ctrlj.es%2Fmayoral%2Fimg%2Fpolo-azul-marino.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fjson.ctrlj.es%2Fmayoral%2Fimg%2Fpolo-azul-marino.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fjson.ctrlj.es%2Fmayoral%2Fimg%2Fpolo-azul-marino.jpg&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </span>
      <h2
        class="productName"
      >
        Polo manga corta estampada
      </h2>
      <div
        class="productCardBody"
      >
        <p
          class="oldPrice"
        >
          18.99
        </p>
        <p
          class="priceWithDiscount"
        >
          15.19 
          <span>
            (-20%)
          </span>
        </p>
        <p
          class="colors"
        >
          más colores
        </p>
      </div>
      <button>
        Añadir
      </button>
    </div>
  </div>
</DocumentFragment>
`;
